{"priority": "major", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/osrf/subt.json"}, "html": {"href": "#!/osrf/subt"}, "avatar": {"href": "data/bytebucket.org/ravatar/{3c95f6ad-c304-407c-b838-09597d836552}ts=2272898"}}, "type": "repository", "name": "subt", "full_name": "osrf/subt", "uuid": "{3c95f6ad-c304-407c-b838-09597d836552}"}, "links": {"attachments": {"href": "data/repositories/osrf/subt/issues/183/attachments_page=1.json"}, "self": {"href": "data/repositories/osrf/subt/issues/183.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/subt/issues/183/watch"}, "comments": {"href": "data/repositories/osrf/subt/issues/183/comments_page=1.json"}, "html": {"href": "#!/osrf/subt/issues/183/rendering-errors-when-running-headless-on"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/subt/issues/183/vote"}}, "reporter": {"display_name": "Sarah Kitchen", "uuid": "{6e33efb0-dd9b-4c54-9597-53cb73099f33}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B6e33efb0-dd9b-4c54-9597-53cb73099f33%7D"}, "html": {"href": "https://bitbucket.org/%7B6e33efb0-dd9b-4c54-9597-53cb73099f33%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/29a4e0df12f5ebcc54cd596a60606526d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsSK-1.png"}}, "nickname": "snkitche", "type": "user", "account_id": "5be108d7216ccf49622e6b67"}, "title": "Rendering errors when running headless on remote machine", "component": null, "votes": 0, "watches": 3, "content": {"raw": "I still haven\u2019t been able to run the simulator remotely without it crashing immediately. When I try to run cloudsim\\_sim \\(via docker\\) with headless:=true, it still crashes with an OGRE2RenderEngine error. \r\n\r\nLooks like this plugin is the culprit in the cloudsim\\_sim.ign file. The other ogre2 engine call is in the block that\u2019s not executed when headless:=true. \r\n\r\n```\r\n    <plugin entity_name=\"<%= $worldName %>\"\r\n            entity_type=\"world\"\r\n            filename=\"libignition-gazebo-sensors-system.so\"\r\n            name=\"ignition::gazebo::systems::Sensors\">\r\n            <render_engine>ogre2</render_engine>\r\n    </plugin>\r\n```\r\n\r\nHere\u2019s some console output at the crash:\r\n\r\n```\r\n[Dbg] [Sensors.cc:336] Initialization needed\r\n[Dbg] [Sensors.cc:170] Initializing render context\r\n[Msg] Loading plugin [ignition-rendering2-ogre2]\r\n[Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread (Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread (Exiting postupdat[Err] [Ogre2RenderEngine.cc:338] Unable to open display: \r\nterminate called after throwing an instance of 'Ogre::RenderingAPIException'\r\n  what():  OGRE EXCEPTION(3:RenderingAPIException): Couldn`t open X display  in GLXGLSupport::getGLDisplay at /var/lib/jenkins/workspace/ogre-2.1-debbuilder/repo/RenderSystems/GL3Plus/src/windowing/GLX/OgreGLXGLSupport.cpp (line 789)\r\n```\r\n\r\n\u200c\r\n\r\nI don\u2019t have the catkin install set up remotely, and I\u2019m not seeing this error locally \\(which does not have a rendering problem and does not crash\\).", "markup": "markdown", "html": "<p>I still haven\u2019t been able to run the simulator remotely without it crashing immediately. When I try to run cloudsim_sim (via docker) with headless:=true, it still crashes with an OGRE2RenderEngine error. </p>\n<p>Looks like this plugin is the culprit in the cloudsim_sim.ign file. The other ogre2 engine call is in the block that\u2019s not executed when headless:=true. </p>\n<div class=\"codehilite\"><pre><span></span>    <span class=\"nt\">&lt;plugin</span> <span class=\"na\">entity_name=</span><span class=\"s\">&quot;</span><span class=\"cp\">&lt;%=</span> <span class=\"vg\">$worldName</span> <span class=\"cp\">%&gt;</span><span class=\"s\">&quot;</span>\n            <span class=\"na\">entity_type=</span><span class=\"s\">&quot;world&quot;</span>\n            <span class=\"na\">filename=</span><span class=\"s\">&quot;libignition-gazebo-sensors-system.so&quot;</span>\n            <span class=\"na\">name=</span><span class=\"s\">&quot;ignition::gazebo::systems::Sensors&quot;</span><span class=\"nt\">&gt;</span>\n            <span class=\"nt\">&lt;render_engine&gt;</span>ogre2<span class=\"nt\">&lt;/render_engine&gt;</span>\n    <span class=\"nt\">&lt;/plugin&gt;</span>\n</pre></div>\n\n\n<p>Here\u2019s some console output at the crash:</p>\n<div class=\"codehilite\"><pre><span></span>[Dbg] [Sensors.cc:336] Initialization needed\n[Dbg] [Sensors.cc:170] Initializing render context\n[Msg] Loading plugin [ignition-rendering2-ogre2]\n[Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread (Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread ([Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] [Dbg] [SimulationRunner.cc:372] Exiting postupdate worker thread (Exiting postupdat[Err] [Ogre2RenderEngine.cc:338] Unable to open display: \nterminate called after throwing an instance of &#39;Ogre::RenderingAPIException&#39;\n  what():  OGRE EXCEPTION(3:RenderingAPIException): Couldn`t open X display  in GLXGLSupport::getGLDisplay at /var/lib/jenkins/workspace/ogre-2.1-debbuilder/repo/RenderSystems/GL3Plus/src/windowing/GLX/OgreGLXGLSupport.cpp (line 789)\n</pre></div>\n\n\n<p>\u200c</p>\n<p>I don\u2019t have the catkin install set up remotely, and I\u2019m not seeing this error locally (which does not have a rendering problem and does not crash).</p>", "type": "rendered"}, "assignee": {"display_name": "Alfredo Bencomo", "uuid": "{84891a71-dff2-4b97-8891-9446c16e6f89}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B84891a71-dff2-4b97-8891-9446c16e6f89%7D"}, "html": {"href": "https://bitbucket.org/%7B84891a71-dff2-4b97-8891-9446c16e6f89%7D/"}, "avatar": {"href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/5cf75629ff02b50ea27b2817/82be1d48-d490-4d7b-8ddb-a8093b9417a6/128"}}, "nickname": "bencomo", "type": "user", "account_id": "5cf75629ff02b50ea27b2817"}, "state": "closed", "version": null, "edited_on": null, "created_on": "2019-09-11T20:38:46.624252+00:00", "milestone": null, "updated_on": "2019-10-31T05:53:41.389374+00:00", "type": "issue", "id": 183}